### ISSUE TITLE
About eval_configs

### ISSUE THREAD

**Issue by MengnanCui** [ACEsuit/mace#30]
**Describe the bug**
Hi, thanks for this great code!
I met an Error report during the evaluation process `RuntimeError: expected scalar type Double but found Float`, however, I believe the training process works very well, since I got RMSE/MAE for both total and each config_type individually, so I am not sure why it happens.

*The full report*
```
Traceback (most recent call last):
  File "/u/mncui/software/mace_v2/scripts/eval_configs.py", line 113, in <module>
    main()
  File "/u/mncui/software/mace_v2/scripts/eval_configs.py", line 78, in main
    output = model(batch, training=False)
  File "/u/mncui/software/anaconda3/envs/mace_env2/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/u/mncui/software/anaconda3/envs/mace_env2/lib/python3.7/site-packages/mace/modules/models.py", line 203, in forward
    node_e0 = self.atomic_energies_fn(data.node_attrs)
  File "/u/mncui/software/anaconda3/envs/mace_env2/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/u/mncui/software/anaconda3/envs/mace_env2/lib/python3.7/site-packages/mace/modules/blocks.py", line 77, in forward
    return torch.matmul(x, self.atomic_energies)
RuntimeError: expected scalar type Double but found Float
```

**To Reproduce**
In order to quickly reproduce this case, I attached the output model, submit scripts for both training and evaluation, and the log file as well.
Thanks so much for any information about it!
[attachments](https://thcloud.rz-berlin.mpg.de/s/pnxYyRXSfPysNF3)

**Comment by MengnanCui** [comment]
[attachments](https://thcloud.rz-berlin.mpg.de/s/pnxYyRXSfPysNF3)

**Comment by davkovacs** [comment]
I think the problem might be cause by setting the defuault datatype in the training script, but not in the evaluation. Can you try setting the flag `--default_dtype` the same way for eval as for train.

**Comment by MengnanCui** [comment]
> I think the problem might be cause by setting the defuault datatype in the training script, but not in the evaluation. Can you try setting the flag `--default_dtype` the same way for eval as for train.

Hey, It works, but with more stories. I set the flag `--default_dtype='float32` firstly, then it gives out the report as below:
```
Traceback (most recent call last):
  File "/u/mncui/software/mace_v2/scripts/eval_configs.py", line 113, in <module>
    main()
  File "/u/mncui/software/mace_v2/scripts/eval_configs.py", line 78, in main
    output = model(batch, training=False)
  File "/u/mncui/software/anaconda3/envs/mace_env2/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/u/mncui/software/anaconda3/envs/mace_env2/lib/python3.7/site-packages/mace/modules/models.py", line 226, in forward
    edge_index=data.edge_index,
  File "/u/mncui/software/anaconda3/envs/mace_env2/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/u/mncui/software/anaconda3/envs/mace_env2/lib/python3.7/site-packages/mace/modules/blocks.py", line 517, in forward
    node_feats[sender], edge_attrs, tp_weights
  File "/u/mncui/software/anaconda3/envs/mace_env2/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/u/mncui/software/anaconda3/envs/mace_env2/lib/python3.7/site-packages/e3nn/o3/_tensor_product/_tensor_product.py", line 529, in forward
    return self._compiled_main_left_right(x, y, real_weight)
  File "/u/mncui/software/anaconda3/envs/mace_env2/lib/python3.7/site-packages/torch/nn/modules/module.py", line 889, in _call_impl
    result = self.forward(*input, **kwargs)
RuntimeError: The following operation failed in the TorchScript interpreter.
Traceback of TorchScript, serialized code (most recent call last):
  File "code/__torch__/torch/fx/graph_module/___torch_mangle_8.py", line 109, in forward
    einsum_18 = torch.einsum("dba,dbc->dbac", [reshape_5, reshape_28])
    mul_2 = torch.mul(reshape_7, 2.2360679774997898)
    tensordot_1 = torch.tensordot(mul_2, _w3j_1_1_2, [2], [1])
                  ~~~~~~~~~~~~~~~ <--- HERE
    einsum_19 = torch.einsum("edab,ecad->ecb", [tensordot_1, einsum_18])
    reshape_29 = torch.reshape(einsum_19, [getitem_4, 640])

Traceback of TorchScript, original code (most recent call last):
    einsum_18 = torch.functional.einsum('dba,dbc->dbac', reshape_5, reshape_28);  reshape_28 = None
    mul_2 = reshape_7 * 2.23606797749979;  reshape_7 = None
    tensordot_1 = torch.functional.tensordot(mul_2, _w3j_1_1_2, [2], [1]);  mul_2 = _w3j_1_1_2 = None
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
    einsum_19 = torch.functional.einsum('edab,ecad->ecb', tensordot_1, einsum_18);  tensordot_1 = einsum_18 = None
    reshape_29 = einsum_19.reshape(getitem_4, 640);  einsum_19 = None
RuntimeError: Tensor for 'out' is on CPU, Tensor for argument #1 'self' is on CPU, but expected them to be on GPU (while checking arguments for addmm)
```
I guess it might be related to cuda setting, so I add another flag `--device=cuda \`, then It works!!!
I checked the `output.xyz`, exactly right, thanks a lot for the information!!!

By the way, another small curiousity about the flag `--config_type_weights`, my understanding, it is a `dict`, where I can define the weight for each config_type like {'diamond':0.5, 'graphene':0.7, ...}, right? 
I wonder is it possible to set a weight for each structure in order to emphasize where could MACE pay attention to (new configurations).

**Comment by davkovacs** [comment]
I am glad it works! :) 
The per configuration weights are implemented in this pull request: https://github.com/ACEsuit/mace/pull/22

I think we will merge it later today to the develop branch. cc @ilyes319

**Comment by MengnanCui** [comment]
Great! Thanks a lot!

**Comment by ilyes319** [comment]
I will close this as we have merged the PR #22.
